template/src/pageComponents/mint/ContractDemo.tsx

```diff
@@ -23,18 +36,21 @@ export default function MintContractDemo() {
     contract.abi,
   );
 
-  const { config } = usePrepareContractWrite({
-    address: contract.status === 'ready' ? contract.address : undefined,
-    abi: contract.abi,
-    functionName: 'mint',
-    args: address ? [address, BigInt(1), BigInt(1), address] : undefined,
-    enabled: onCorrectNetwork,
-  });
+  const mintContent = useMemo(() => {
+    if (mintStep === MintSteps.MINT_PROCESSING_STEP) {
+      return <MintProcessingStep />;
+    }
+
+    if (mintStep === MintSteps.OUT_OF_GAS_STEP) {
```

Comment:
It's a good question, and something I was thinking a lot about. I decided against abstracting them even though there is similarity due to the fact that we may have complex logic in these steps at some point that are specific to the particular action (Mint, Buy Coffee, etc.). If we continue to see overlap it may be worth abstracting in a future PR.